internal/services/ingestor/ingestor.go

```diff
@@ -92,14 +102,18 @@ func NewIngestor(fs ...IngestorOptFunc) (Ingestor, error) {
 	}
 
 	return &IngestorImpl{
-		eventRepository:       opts.eventRepository,
-		streamEventRepository: opts.streamEventRepository,
-		logRepository:         opts.logRepository,
-		mq:                    opts.mq,
-		v:                     validator.NewDefaultValidator(),
+		eventRepository:        opts.eventRepository,
+		streamEventRepository:  opts.streamEventRepository,
+		entitlementsRepository: opts.entitlementsRepository,
+
+		logRepository: opts.logRepository,
+		mq:            opts.mq,
+		v:             validator.NewDefaultValidator(),
 	}, nil
 }
 
+var ErrResourceExhausted = fmt.Errorf("resource exhausted: cannot create event for tenant at this time")
```

Comment:
Why do we have a new `ErrResourceExhausted` within the service instead of using a shared error?

Reply:
good catch